pkg/controllers/secrets_test.go

```diff
@@ -0,0 +1,123 @@
+	name                         string
+	secrets                      map[string]string
+	expectedDangerousSecretNames []string
+}
+
+func TestCheckForDangerousSecretNames(t *testing.T) {
+	testCases := []dangerousSecretNameTestCase{
+		{
+			name: "Should not find any dangerous secret names",
+			secrets: map[string]string{
+				"MY_SECRET": "123",
+			},
+			expectedDangerousSecretNames: nil,
+		},
+		{
+			name: "Should find a dangerous secret name",
+			secrets: map[string]string{
+				"DYLD_INSERT_LIBRARIES": "123",
+				"MY_SECRET":             "123",
+			},
+			expectedDangerousSecretNames: []string{"DYLD_INSERT_LIBRARIES"},
+		},
+		{
+			name: "Should find multiple dangerous secret names",
+			secrets: map[string]string{
+				"DYLD_INSERT_LIBRARIES": "123",
+				"MY_SECRET":             "123",
+				"LD_LIBRARY_PATH":       "123",
+				"WINDIR":                "123",
+				"PROMPT_COMMAND":        "123",
+			},
+			expectedDangerousSecretNames: []string{"DYLD_INSERT_LIBRARIES", "LD_LIBRARY_PATH", "WINDIR", "PROMPT_COMMAND"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			err := CheckForDangerousSecretNames(testCase.secrets)
+			if testCase.expectedDangerousSecretNames == nil {
+				assert.Nil(t, nil, err)
+			} else {
+				for _, v := range testCase.expectedDangerousSecretNames {
+					assert.Contains(t, err.Error(), v)
+				}
+			}
+		})
+	}
+}
+
+type selectSecretsTestCase struct {
+	name         string
+	origMap      map[string]string
+	keysToSelect []string
+	expectedMap  map[string]string
+	missingKeys  []string
+}
+
+func TestSelectSecrets(t *testing.T) {
+	testCases := []selectSecretsTestCase{
+		{
+			name:         "Select one exisiting secret and two nonexistent secrets",
+			origMap:      map[string]string{"MY_SECRET": "value"},
+			keysToSelect: []string{"DEV", "LOGGING", "MY_SECRET"},
+			expectedMap:  map[string]string{"MY_SECRET": "value"},
+			missingKeys:  []string{"DEV", "LOGGING"},
+		},
+		{
+			name:         "Select one secret",
+			origMap:      map[string]string{"DEV": "true", "LOGGING": "true"},
+			keysToSelect: []string{"DEV"},
+			expectedMap:  map[string]string{"DEV": "true"},
+		},
+		{
+			name:         "Select multiple secrets",
+			origMap:      map[string]string{"DEV": "true", "LOGGING": "true", "MY_SECRET": "value", "PROD": "false"},
+			keysToSelect: []string{"DEV", "LOGGING", "PROD"},
+			expectedMap:  map[string]string{"DEV": "true", "LOGGING": "true", "PROD": "false"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			filteredSecrets, err := SelectSecrets(testCase.origMap, testCase.keysToSelect)
+			if testCase.missingKeys != nil {
+				assert.NotNil(t, err)
+				for _, missingKey := range testCase.missingKeys {
+					assert.Contains(t, err.Error(), missingKey)
+				}
+			} else {
+				assert.Nil(t, err)
+				assert.True(t, reflect.DeepEqual(filteredSecrets, testCase.expectedMap))
```

Comment:
I think we want to perform this assertion in both cases, right? (i.e. after the whole if-else) `filteredSecrets` will be returned regardless of whether or not there's a missing key error

Reply:
You are very correct @nmanoogian! Thank you